### PR TITLE
Enforce secure transport policy on artifacts bucket

### PR DESCRIPTION
**Pull request by vladtsir**
*Issue #, if available:* n/a

*Description of changes:*

Enforce "secure transport" bucket policy on the artifacts bucket created in the user account

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jotompki** [comment]
Have you tested end to end submitting a resource type? Want to make sure registration still works with this condition

File: src/rpdk/core/data/managed-upload-infrastructure.yaml [hunk 1]
```diff
@@ -38,6 +38,24 @@ Resources:
               - !Sub "arn:${AWS::Partition}:s3:::${ArtifactBucket}"
               - !Sub "arn:${AWS::Partition}:s3:::${ArtifactBucket}/*"
 
+  ArtificatBucketSecureCopyPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref ArtifactBucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
```
Comment (jotompki) [inline src/rpdk/core/data/managed-upload-infrastructure.yaml]: This could also be added to the above bucket policy as another statement

Comment (vladtsir) [reply src/rpdk/core/data/managed-upload-infrastructure.yaml]: Folded "secure transport" policy into ArtifactCopyPolicy.

**Comment by vladtsir** [comment]
> Have you tested end to end submitting a resource type? Want to make sure registration still works with this condition

Yes, I was able to perform `cfn submit` and confirmed that the bucket policy was updated

**Pull request merged**